### PR TITLE
remove_occupation

### PR DESCRIPTION
**Pull request by peisenha**
We want to start our development by restricting the model to only one occupation. In this process we also set up our sound software engineering infrastructure.

- [x] remove all traces of Occupation B in the code
- [x] run the indices for `period` and `type` from one instead of zero to align with the FORTRAN indexing.
- [x] extensive exploratory testing
- [x] set up basic testing infrastructure such as regression tests and property testing

### CODE REVIEW

File: norpy/src/lib_norpy.f90 [hunk 1]
```diff
@@ -25,22 +25,22 @@ MODULE lib_norpy
     REAL(our_dble), PARAMETER :: two_dble = 2.00_our_dble
 
     TYPE COVARIATES_DICT
-
+        !Here I deleted all references to occupation B 
         INTEGER(our_int) :: is_return_not_high_school
         INTEGER(our_int) :: is_return_high_school
         INTEGER(our_int) :: not_exp_a_lagged
```
Comment (peisenha) [inline norpy/src/lib_norpy.f90]: Please remove all references to an occupation A from all variable names throughout the package.

File: norpy/src/lib_norpy.f90 [hunk 2]
```diff
@@ -25,22 +25,22 @@ MODULE lib_norpy
...
-        INTEGER(our_int) :: not_exp_b_lagged
+
         INTEGER(our_int) :: is_young_adult
         INTEGER(our_int) :: choice_lagged
         INTEGER(our_int) :: work_a_lagged
         INTEGER(our_int) :: work_b_lagged
```
Comment (peisenha) [inline norpy/src/lib_norpy.f90]: Please remove all objects that were only used for occupation B.

File: norpy/src/lib_norpy.f90 [hunk 3]
```diff
@@ -56,24 +56,25 @@ MODULE lib_norpy
 CONTAINS
     !***********************************************************************************************
     !***********************************************************************************************
-    FUNCTION calculate_wages_systematic(covariates, coeffs_a, coeffs_b, type_shifts) RESULT(wages)
+    FUNCTION calculate_wages_systematic(covariates, coeffs_a, type_shifts) RESULT(wages)
 
         !/* dummy arguments        */
+	!Here I changed all the references to occ_b
```
Comment (peisenha) [inline norpy/src/lib_norpy.f90]: Please clean up / remove all comments,

File: norpy/src/lib_norpy.f90 [hunk 4]
```diff
@@ -56,24 +56,25 @@ MODULE lib_norpy
...
+    FUNCTION calculate_wages_systematic(covariates, coeffs_a, type_shifts) RESULT(wages)
 
         !/* dummy arguments        */
+	!Here I changed all the references to occ_b 
 
-        REAL(our_dble) :: wages(2)
+        REAL(our_dble) :: wages
 
         REAL(our_dble), INTENT(IN) :: type_shifts(:, :)
         REAL(our_dble), INTENT(IN) :: coeffs_a(15)
```
Comment (peisenha) [inline norpy/src/lib_norpy.f90]: This is the reason for the compiler warning, as `coeffs_a`only has 13 elements now. Please fix.

File: norpy/src/lib_norpy.f90 [hunk 5]
```diff
@@ -83,31 +84,29 @@ FUNCTION calculate_wages_systematic(covariates, coeffs_a, coeffs_b, type_shifts)
...
+        wages = wages * EXP(type_shifts(covariates%type + 1,1))
+        
 
     END FUNCTION
     !***********************************************************************************************
     !***********************************************************************************************
     FUNCTION calculate_rewards_common(covariates, coeffs_common) RESULT(rewards_common)
 
         !/* dummy arguments       */
+
```
Comment (peisenha) [inline norpy/src/lib_norpy.f90]: Please make sure we do not have too many blank lines in the function/subroutine headers.

File: norpy/src/lib_norpy.f90 [hunk 6]
```diff
@@ -129,30 +128,29 @@ FUNCTION calculate_rewards_common(covariates, coeffs_common) RESULT(rewards_comm
     END FUNCTION
     !***********************************************************************************************
     !***********************************************************************************************
-    FUNCTION calculate_rewards_general(covariates, coeffs_a_general, coeffs_b_general) RESULT(rewards_general)
+    FUNCTION calculate_rewards_general(covariates, coeffs_a_general) RESULT(rewards_general)
 
         !/* dummy arguments        */
-
-        REAL(our_dble) :: rewards_general(2)
+        !Why no INTENT(OUT) ??????
```
Comment (peisenha) [inline norpy/src/lib_norpy.f90]: We discussed this, but please be sure to confirm my explanation.

File: norpy/src/norpy_hatchery.f90 [hunk 7]
```diff
@@ -60,56 +61,58 @@ SUBROUTINE f2py_create_state_space(states_all, states_number_period, mapping_sta
...
+                                
                                 ! ! If we have multiple initial conditions it might well be the case that we have a duplicate state, i.e. the same state is possible with other initial condition that period.
-                                IF (mapping_state_idx(period + 1, exp_a + 1, exp_b + 1, edu_start + edu_add + 1, choice_lagged, type_ + 1) .NE. MISSING_INT) CYCLE
+
+                                IF (mapping_state_idx(period, exp_a+1 ,edu_start+ edu_add+1 , choice_lagged, type_ ) .NE. MISSING_INT) CYCLE
+                                
 
                                 ! ! Collect mapping of state space to array index.
-                                mapping_state_idx(period + 1, exp_a + 1, exp_b + 1, edu_start + edu_add + 1, choice_lagged, type_ + 1) = k
+                                mapping_state_idx(period , exp_a+1 , edu_start+edu_add+1 , choice_lagged, type_ ) = k
```
Comment (peisenha) [inline norpy/src/norpy_hatchery.f90]: Let us be consistent about the spaces before and after operators as in Python's PEP8.

File: norpy/src/norpy_hatchery.f90 [hunk 8]
```diff
@@ -223,31 +236,35 @@ SUBROUTINE f2py_calculate_immediate_rewards(periods_rewards_systematic, num_peri
...
+            rewards(2) = DOT_PRODUCT(covars_edu, coeffs_edu)
 
             ! Calculate systematic part of HOME
             covars_home(1) = one_int
+            
             covars_home(2) = covariates%is_young_adult
-            covars_home(3) = covariates%period - one_int
-
-            rewards(4) = DOT_PRODUCT(covars_home, coeffs_home)
+            covars_home(3) = covariates%period - one_int ! Ok do I have to modify this for the new index ?
```
Comment (peisenha) [inline norpy/src/norpy_hatchery.f90]: Please align this variable with the Keane & Wolpin's equations in their 1997 paper Appendix.

File: norpy/src/lib_norpy.f90 (hunk 1 above)
Comment (mo2561057) [reply norpy/src/lib_norpy.f90]: done

File: norpy/src/lib_norpy.f90 (hunk 2 above)
Comment (mo2561057) [reply norpy/src/lib_norpy.f90]: done

File: norpy/src/lib_norpy.f90 (hunk 3 above)
Comment (mo2561057) [reply norpy/src/lib_norpy.f90]: All the comments that I added or in general ?

File: norpy/src/lib_norpy.f90 (hunk 4 above)
Comment (mo2561057) [reply norpy/src/lib_norpy.f90]: done

File: norpy/src/lib_norpy.f90 (hunk 3 above)
Comment (peisenha) [reply norpy/src/lib_norpy.f90]: In general, I was using them way too much. This is mostly your code base now, so please just keep the ones you deem useful.

**Pull request merged**